samplers/samplers.go

```diff
@@ -21,54 +21,91 @@ const RedshiftDateFormat = "2006-01-02 03:04:05"
 type DDMetric struct {
 	Name       string        `json:"metric"`
 	Value      [1][2]float64 `json:"points"`
-	Tags       []string      `json:"tags,omitempty"`
+	Tags       []string      `json:"Tags,omitempty"`
 	MetricType string        `json:"type"`
 	Hostname   string        `json:"host"`
-	DeviceName string        `json:"device_name"`
+	DeviceName string        `json:"device_Name"`
 	Interval   int32         `json:"interval,omitempty"`
 }
 
+// EncodeCSV generates a newline-terminated CSV row that describes
+// the data represented by the DDMetric.
+// The caller is responsible for setting w.Comma as the appropriate delimiter.
+// For performance, encodeCSV does not flush after every call; the caller is
+// expected to flush at the end of the operation cycle
+func (d DDMetric) EncodeCSV(w *csv.Writer, partitionDate *time.Time, hostName string) error {
```

Comment:
hm now that we've factored samplers into a separate package, i have mixed feelings on this living here, since stuff like partition date is an implementation detail of s3 upload in particular. wdyt? could/should this live elsewhere, particularly in the s3 package?